lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
@serbanmaerean  : I do use `xlc` and `xlc_r` versions on BG-Q (but not fortran).  Would it be sufficient to use different `spec` in `compilers.yaml`? something like : 
```
compilers:
- compiler:
    modules: [bg-xl]
    operating_system: cnk
    paths:
      cc: /opt/ibmcmp/vacpp/bg/12.1/bin/bgxlc
      cxx: /opt/ibmcmp/vacpp/bg/12.1/bin/bgxlc++
      f77: /opt/ibmcmp/xlf/bg/14.1/bin/xlf
      fc: /opt/ibmcmp/xlf/bg/14.1/bin/xlf2008
    spec: xl@12.1
- compiler:
    modules: [bg-xl]
    operating_system: cnk
    paths:
      cc: /opt/ibmcmp/vacpp/bg/12.1/bin/bgxlc_r
      cxx: /opt/ibmcmp/vacpp/bg/12.1/bin/bgxlc++_r
      f77: /opt/ibmcmp/xlf/bg/14.1/bin/xlf_r
      fc: /opt/ibmcmp/xlf/bg/14.1/bin/xlf2008_r
    spec: xl@12.r.1       # OR some other name
```